custom_components/solaredge_modbus/__init__.py

```diff
@@ -829,59 +824,59 @@ def decode_string(decoder):
-            battery_availbable = decoder.decode_32bit_float()
-            #0x82 - 2 - SoH %
-            battery_SoH = decoder.decode_32bit_float()
-            #0x84 - 2 - SoC %
-            battery_SoC = validate(decoder.decode_32bit_float(), ">", 0.0)
-            battery_SoC = validate(battery_SoC, "<", 101)
-
-            self.data[battery_prefix + 'temp_avg'] = round(tempavg, 1)
-            self.data[battery_prefix + 'temp_max'] = round(tempmax, 1)
-            self.data[battery_prefix + 'voltage'] = round(batteryvoltage, 3)
-            self.data[battery_prefix + 'current'] = round(batterycurrent, 3)
-            self.data[battery_prefix + 'power'] = round(batterypower, 3)
-            self.data[battery_prefix + 'energy_discharged'] = round(cumulative_discharged / 1000, 3)
-            self.data[battery_prefix + 'energy_charged'] = round(cumulative_charged / 1000, 3)
-            self.data[battery_prefix + 'size_max'] = round(battery_max, 3)
-            self.data[battery_prefix + 'size_available'] = round(battery_availbable, 3)
-            self.data[battery_prefix + 'state_of_health'] = round(battery_SoH, 0)
-            self.data[battery_prefix + 'state_of_charge'] = round(battery_SoC, 0)
-            battery_status = decoder.decode_32bit_uint()
-
-            # voltage and current are bogus in certain statuses
-            if not battery_status in [3,4,6]:
-                self.data[battery_prefix + 'voltage'] = 0
-                self.data[battery_prefix + 'current'] = 0
-                self.data[battery_prefix + 'power'] = 0
-
-            if battery_status in BATTERY_STATUSSES:
-                self.data[battery_prefix + 'status'] = BATTERY_STATUSSES[battery_status]
-            else:
-                self.data[battery_prefix + 'status'] = battery_status
+        decoder = BinaryPayloadDecoder.fromRegisters(
+            storage_data.registers, byteorder=Endian.Big,wordorder=Endian.Little
+        )
 
-            return True
+        #0x6C - 2 - avg temp C
+        tempavg = decoder.decode_32bit_float()
+        #0x6E - 2 - max temp C
+        tempmax = decoder.decode_32bit_float()
+        #0x70 - 2 - inst voltage V
+        batteryvoltage = decoder.decode_32bit_float()
+        #0x72 - 2 - inst current A
+        batterycurrent = decoder.decode_32bit_float()
+        #0x74 - 2 - inst power W
+        batterypower = decoder.decode_32bit_float()
+        #0x76 - 4 - cumulative discharged (Wh)
+        cumulative_discharged = decoder.decode_64bit_uint()
+        #0x7a - 4 - cumulative charged (Wh)
+        cumulative_charged = decoder.decode_64bit_uint()
+        #0x7E - 2 - current max size Wh
+        battery_max = decoder.decode_32bit_float()
+        #0x80 - 2 - available size Wh
+        battery_availbable = decoder.decode_32bit_float()
+        #0x82 - 2 - SoH %
+        battery_SoH = decoder.decode_32bit_float()
+        #0x84 - 2 - SoC %
+        battery_SoC = validate(decoder.decode_32bit_float(), ">", 0.0)
```

Comment:
I do think this should be `battery_SoC = validate(decoder.decode_32bit_float(), ">=", 0.0)` as i'd have thought `0.0` to be valid.